logbook-servlet/src/main/java/org/zalando/logbook/servlet/LocalResponse.java

```diff
@@ -106,42 +114,54 @@ public PrintWriter getWriter() throws IOException {
         return body;
     }
 
-    private final class TeeServletOutputStream extends ServletOutputStream {
+    private static final class TeeServletOutputStream extends ServletOutputStream {
 
-        private final OutputStream original;
+        private final DataOutput output;
+        private final LazyStream original;
 
-        private TeeServletOutputStream() throws IOException {
-            this.original = LocalResponse.super.getOutputStream();
+        private TeeServletOutputStream(final DataOutput output,
+                final LazyStream original) {
```

Comment:
Also I think current implementation does not solve the core issue, i.e. getOutputStream could be called multiple times.

I would propose:
```diff
--- a/logbook-servlet/src/main/java/org/zalando/logbook/servlet/LocalResponse.java
+++ b/logbook-servlet/src/main/java/org/zalando/logbook/servlet/LocalResponse.java
@@ -40,7 +40,7 @@ final class LocalResponse extends HttpServletResponseWrapper implements RawHttpR
 
     LocalResponse(final HttpServletResponse response, final String protocolVersion) throws IOException {
         super(response);
-        this.stream = new TeeServletOutputStream(output, super::getOutputStream);
+        this.stream = new TeeServletOutputStream(output, new LazyResponseOutputStream(response));
         this.writer = new TeePrintWriter(stream, getCharset());
         this.protocolVersion = protocolVersion;
     }
@@ -116,46 +116,76 @@ final class LocalResponse extends HttpServletResponseWrapper implements RawHttpR
 
     private static final class TeeServletOutputStream extends ServletOutputStream {
 
-        private final DataOutput output;
-        private final LazyStream original;
+        private final OutputStream first;
+        private final OutputStream second;
 
-        private TeeServletOutputStream(final DataOutput output,
-                final LazyStream original) {
-            this.output = output;
-            this.original = original;
+        private TeeServletOutputStream(final OutputStream first, final OutputStream second) {
+            this.first = first;
+            this.second = second;
         }
 
         @Override
         public void write(final int b) throws IOException {
-            output.write(b);
-            original().write(b);
+            first.write(b);
+            second.write(b);
         }
 
         @Override
         public void write(final byte[] b, final int off, final int len) throws IOException {
-            output.write(b, off, len);
-            original().write(b, off, len);
+            first.write(b, off, len);
+            second.write(b, off, len);
         }
 
         @Override
         public void flush() throws IOException {
-            original().flush();
+            first.flush();
+            second.flush();
         }
 
         @Override
         public void close() throws IOException {
-            original().close();
+            first.close();
+            second.close();
         }
 
-        private OutputStream original() throws IOException {
-            return original.retrieveLazily();
+    }
+
+    private static final class LazyResponseOutputStream extends OutputStream {
+
+        private final HttpServletResponse response;
+        private OutputStream stream;
+
+        LazyResponseOutputStream(final HttpServletResponse response) {
+            this.response = response;
         }
 
-    }
+        @Override
+        public void write(final int b) throws IOException {
+            getStream().write(b);
+        }
+
+        @Override
+        public void write(final byte[] b, final int off, final int len) throws IOException {
+            getStream().write(b, off, len);
+        }
+
+        @Override
+        public void flush() throws IOException {
+            getStream().flush();
+        }
+
+        @Override
+        public void close() throws IOException {
+            getStream().close();
+        }
+
+        private OutputStream getStream() throws IOException {
+            if (stream == null) {
+                stream = response.getOutputStream();
+            }
+            return stream;
+        }
 
-    @FunctionalInterface
-    public interface LazyStream {
-        ServletOutputStream retrieveLazily() throws IOException;
     }
 
     private static final class TeePrintWriter extends PrintWriter {

```